third_party/tonic/dart_state.cc

```diff
@@ -32,8 +33,15 @@ DartState::~DartState() {}
 
 void DartState::SetIsolate(Dart_Isolate isolate) {
   isolate_ = isolate;
+
   if (!isolate_)
     return;
+
+  private_constructor_name_.Clear();
+  Dart_EnterScope();
+  private_constructor_name_.Set(this, Dart_NewPersistentHandle(Dart_NewStringFromCString("_")));
```

Comment:
Is there any reason why the string "_" can't be cached rather than creating it for each `DartState`?

Reply:
This is tieing it to the lifetime of the isolate.  We don't want to leak it if the isolate changes.